File: src/Entities/Customer.cs
using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.AspNetCore.Identity.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore;
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
// using Newtonsoft.Json;

namespace Intertel.Entities
{
  public class Customer
  {
    [Key]
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string CompanyName { get; set; }
    public string Email { get; set; }
    public int ContactNo { get; set; }
    public string Address { get; set; }
    public IdentityUser User { get; set; }
    [DataType(DataType.Date)]
    [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
    public DateTime? CreatedAt { get; set; } = DateTime.Now;
    [DataType(DataType.Date)]
    [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
    public DateTime? UpdatedAt { get; set; } = DateTime.Now;

  }
}

File: src/ts/models/customer.ts
import {Item} from "./item"

export class Customer {
  id: string = "";
  name: string = "";
  companyName: string = "";
  email: string = "";
  contactNo: number = 0;
  address: string = "";
  userId: string = "";
  createdAt: Date = new Date();
  updatedAt:Date = new Date();
  sortNo: number = 1;
  Items:Item[]|null = [];
 }